spring-cloud-services-config-client-autoconfigure/src/main/java/io/pivotal/spring/cloud/config/client/PlainTextConfigClientImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pivotal.spring.cloud.config.client;
+
+import org.springframework.cloud.config.client.ConfigClientProperties;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * {@link RestTemplate} based implementation of {@link PlainTextConfigClient}.
+ * Config Server URI, default application name, profiles and labels are provided by
+ * {@link ConfigClientProperties}.
+ * 
+ * @author Daniel Lavoie
+ * @author Dylan Roberts
+ */
+class PlainTextConfigClientImpl implements PlainTextConfigClient {
+	private final ConfigClientProperties configClientProperties;
+	private final RestTemplate restTemplate;
+
+	PlainTextConfigClientImpl(RestTemplate restTemplate,
+							  ConfigClientProperties configClientProperties) {
+		this.restTemplate = restTemplate;
+		this.configClientProperties = configClientProperties;
+	}
+
+	/**
+	 * Retrieves a config file using the defaults application name, profiles and labels.
+	 * 
+	 * @throws IllegalArgumentException when application name or Config Server url is
+	 * undefined.
+	 * @throws HttpClientErrorException when a config file is not found.
+	 */
+	@Override
+	public Resource getConfigFile(String path) {
+		return getConfigFile(null, null, path);
+	}
+
+	/**
+	 * Retrieves a config file.
+	 * 
+	 * @throws IllegalArgumentException when application name or Config Server url is
+	 * undefined.
+	 * @throws HttpClientErrorException when a config file is not found.
+	 */
+	@Override
+	public Resource getConfigFile(String profile, String label, String path) {
+		Assert.isTrue(!StringUtils.isEmpty(configClientProperties.getName()),
```

Comment:
Couldn't we use an Assert.notEmpty like the one below this?

Reply:
Unfortunately, there's no `notEmpty` for strings.

Reply:
ahhh. understood. maybe change it a bit to remove the `!` logic, so it's more easy to read. e.g. use `StringUtils.hasLength`, or `StringUtils.hasText`?